### PR TITLE
Minor memory optimizations

### PR DESCRIPTION
**Pull request by tas50**
Shave a bit more memory usage off Ohai

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: lib/ohai/plugins/elixir.rb [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 Ohai.plugin(:Elixir) do
   provides "languages/elixir"
```
Comment (jaymzh) [inline lib/ohai/plugins/elixir.rb:17]: can probably freeze this one too, no?

File: lib/ohai/plugins/go.rb [hunk 2]
```diff
@@ -15,7 +15,7 @@
 
 Ohai.plugin(:Go) do
   provides "languages/go"
```
Comment (jaymzh) [inline lib/ohai/plugins/go.rb:17]: freeze?

**Review by jaymzh — APPROVED** [review]
You freeze one `provides`, but not the rest... feels like they can all be frozen.

File: lib/ohai/plugins/elixir.rb (hunk 1 above)
Comment (tas50) [reply lib/ohai/plugins/elixir.rb:17]: That wasn't showing up in the retained objects. The langs string was in there 10 times

**Pull request merged**